src/invariant/BucketListIsConsistentWithDatabaseTests.cpp

```diff
@@ -495,187 +442,193 @@ class ApplyBucketsWorkModifyEntry : public ApplyBucketsWork
-}
-
-TEST_CASE("BucketListIsConsistentWithDatabase test non-root account",
-          "[invariant][bucketlistconsistent]")
-{
-    VirtualClock clock;
-    Application::pointer appGenerate =
-        createTestApplication(clock, getTestConfig(0));
-    Application::pointer appApply =
-        createTestApplication(clock, getTestConfig(1));
-    AccountID account;
-    uint32_t ledgerSeq =
-        generateLedgers(appGenerate, 2, 1, 1,
-                        [appGenerate, &account](
-                            uint32_t n) -> std::vector<EntryFrame::pointer> {
-                            LedgerEntry entry;
-                            entry.data.type(ACCOUNT);
-                            entry.data.account() =
-                                LedgerTestUtils::generateValidAccountEntry(5);
-                            account = entry.data.account().accountID;
-                            return {EntryFrame::FromXDR(entry)};
-                        });
-    REQUIRE_NOTHROW(
-        applyBucketsAndCrankUntilDone(appGenerate, appApply, ledgerSeq));
-
-    ledgerSeq = generateLedgers(
-        appGenerate, ++ledgerSeq, 100, 1,
-        [appGenerate, account](uint32_t n) -> std::vector<EntryFrame::pointer> {
-            auto& db = appGenerate->getDatabase();
-            auto af = AccountFrame::loadAccount(account, db);
-            return {af};
-        });
-    REQUIRE_NOTHROW(
-        applyBucketsAndCrankUntilDone(appGenerate, appApply, ledgerSeq));
+    BucketListGenerator blg;
+    blg.generateLedgers(100);
+    REQUIRE_NOTHROW(blg.applyBuckets());
+    blg.generateLedgers(100);
+    REQUIRE_NOTHROW(blg.applyBuckets());
+    blg.generateLedgers(100);
+    REQUIRE_NOTHROW(blg.applyBuckets());
 }
 
 TEST_CASE("BucketListIsConsistentWithDatabase test root account",
           "[invariant][bucketlistconsistent]")
 {
-    VirtualClock clock;
-    Application::pointer appGenerate =
-        createTestApplication(clock, getTestConfig(0));
-    Application::pointer appApply =
-        createTestApplication(clock, getTestConfig(1));
-    uint32_t ledgerSeq = generateLedgers(appGenerate, 2, 1, 1);
-    ledgerSeq = generateLedgers(
-        appGenerate, ++ledgerSeq, 100, 1,
-        [appGenerate](uint32_t n) -> std::vector<EntryFrame::pointer> {
-            auto skey = SecretKey::fromSeed(appGenerate->getNetworkID());
-            auto root = skey.getPublicKey();
-            auto& db = appGenerate->getDatabase();
-            auto af = AccountFrame::loadAccount(root, db);
-            return {af};
-        });
-    REQUIRE_NOTHROW(
-        applyBucketsAndCrankUntilDone(appGenerate, appApply, ledgerSeq));
+    struct TestRootBucketListGenerator : public BucketListGenerator
+    {
+        uint32_t const mTargetLedger;
+        bool mModifiedRoot;
+
+        TestRootBucketListGenerator()
+            : BucketListGenerator()
+            , mTargetLedger(
+                  std::uniform_int_distribution<size_t>(2, 100)(*mGen))
```

Comment:
should be `uint32_t` not `size_t` (`mTargetLedger`):
conversion from 'size_t' to 'const uint32_t', possible loss of data